File: Ch01/ex1_04.rb
#!/usr/bin/env ruby
# ex1_04.rb

=begin rdoc
Kevin C. Baird
SICP in Ruby
Demonstrate an operator that is a compound expression
(i.e. not known at compile time).
=end

def a_plus_abs_b(a, b)
  op = b > 0 ? :+ : :-
  a.send(op, b)
end

File: Ch01/Ex1_04.hs
{-
 - Ex1_04.hs
 - Kevin C. Baird
 - SICP in Haskell
 -
 - Demonstrate an operator that is a compound expression
 - (i.e., not known at compile time).
 -}

aPlusAbsB :: Int -> Int -> Int
aPlusAbsB a b = a `op` b
  where
    op
      | b > 0     = (+)
      | otherwise = (-)
